xknx/devices/climate_mode.py

```diff
@@ -218,55 +253,108 @@ async def _set_internal_operation_mode(self, operation_mode):
                 await rv.set(operation_mode)
 
         await self._set_internal_operation_mode(operation_mode)
 
+    async def set_controller_mode(self, controller_mode):
+        """Set the controller mode of a thermostat. Send new controller mode to the bus and update internal state."""
+        if (
+            not self.supports_controller_mode
+            or controller_mode not in self._controller_modes
+        ):
+            raise DeviceIllegalValue(
+                "controller (HVAC) mode not supported", controller_mode
+            )
+
+        for rv in self._iter_controller_remote_values():
+            if rv.writable and controller_mode in rv.supported_operation_modes():
+                await rv.set(controller_mode)
+
+        await self._set_internal_controller_mode(controller_mode)
+
     @property
     def operation_modes(self):
         """Return all configured operation modes."""
         if not self.supports_operation_mode:
             return []
-        return self.operation_modes_
+        return self._operation_modes
+
+    @property
+    def controller_modes(self):
+        """Return all configures controller modes."""
+        if not self.supports_controller_mode:
+            return []
+        return self._controller_modes
 
     def gather_operation_modes(self):
         """Gather operation modes from RemoteValues."""
         operation_modes = []
-        for rv in self._iter_remote_values():
+        for rv in self._iter_binary_remote_values():
+            if rv.writable:
+                operation_modes.extend(rv.supported_operation_modes())
+
+        for rv in self._iter_normal_operation_modes():
+            if rv.writable:
+                operation_modes.extend(rv.supported_operation_modes())
+
+        # remove duplicates
+        return list(set(operation_modes))
+
+    def gather_controller_modes(self):
+        """Gather controller modes from RemoteValues."""
+        operation_modes = []
+        for rv in self._iter_controller_remote_values():
             if rv.writable:
                 operation_modes.extend(rv.supported_operation_modes())
+
         # remove duplicates
         return list(set(operation_modes))
 
     async def process_group_write(self, telegram):
         """Process incoming and outgoing GROUP WRITE telegram."""
         if self.supports_operation_mode:
-            processed = False
             for rv in self._iter_remote_values():
                 if await rv.process(telegram):
-                    # don't set when binary climate mode rv is False
                     if rv.value:
                         await self._set_internal_operation_mode(rv.value)
-                        return
-                    processed = True
-            # if no operation mode has been set and all binary operation modes are False
-            if processed:
-                await self._set_internal_operation_mode(HVACOperationMode.STANDBY)
+
```

Comment:
```suggestion
                    return
```
When a telegram is processed we can stop iterating - a GA should not be used in multiple RemoteValues here in any way.